conda_forge_tick/update_sources.py

```diff
@@ -651,80 +653,125 @@ class NVIDIA(AbstractSource):
+            "libnvjpeg_2k": {  # this is the key which corresponds to redist-json-name
+                "name": "NVIDIA nvJPEG 2000",
+                "license": "nvJPEG 2K",
+                "license_path": "libnvjpeg_2k/LICENSE.txt",
+                "version": "0.8.1.40",
+                ...
+        }
+        ```
+
+        Each of "compute-subdir" and "redist-json-name" will fallback to using the
+        "feedstock-name" parameter if undefined.
+
+        The returned url is not actually valid since we return the json slug that we need
+        by separating it with #.
+        """
         url = meta_yaml["url"]
-        if "nvidia.com" not in url:
+        if "developer.download.nvidia.com/compute" not in url:
+            logger.debug(
+                "The source URL did not contain developer.download.nvidia.com/compute."
+            )
             return None
-        feedstock_name = meta_yaml["feedstock_name"]
-        name = self.feedstock_to_package.get(feedstock_name, feedstock_name)
-        # we need major.minor.patch
-        current_ver = meta_yaml["version"]
-        if current_ver.count(".") > 2:
-            current_ver = current_ver.split(".")
-            current_ver = ".".join(current_ver[:3])
-        elif current_ver.count(".") == 1:
-            current_ver = f"{current_ver}.0"
-        return self.next_ver_func(name, current_ver)
+        logger.debug("The source URL contains deveoper.download.nvidia.com/compute.")
+
+        logger.debug("Searching for a NVIDIA URL compute-subdir.")
+        if "compute-subdir" in meta_yaml["meta_yaml"]["extra"]:
+            logger.debug("Found explicit extra/compute-subdir.")
+            nvidia_compute_subdir = meta_yaml["meta_yaml"]["extra"]["compute-subdir"]
+        elif "feedstock-name" in meta_yaml["meta_yaml"]["extra"]:
+            logger.debug("Found extra/feedstock-name.")
+            nvidia_compute_subdir = meta_yaml["meta_yaml"]["extra"]["feedstock-name"]
+        else:
+            logger.debug("Found the feedstock's name.")
+            nvidia_compute_subdir = meta_yaml["feedstock_name"]
+        logger.debug("The compute-subdir for NVIDIA URL is %s.", nvidia_compute_subdir)
+
+        logger.debug("Searching for a NVIDIA URL redist-json-name.")
+        if "redist-json-name" in meta_yaml["meta_yaml"]["extra"]:
+            logger.debug("Found explicit extra/compute-subdir.")
+            nvidia_redist_json_name = meta_yaml["meta_yaml"]["extra"][
+                "redist-json-name"
+            ]
+        elif "feedstock-name" in meta_yaml["meta_yaml"]["extra"]:
+            logger.debug("Found extra/feedstock-name.")
+            nvidia_redist_json_name = meta_yaml["meta_yaml"]["extra"]["feedstock-name"]
+        else:
+            logger.debug("Found the feedstock's name.")
+            nvidia_redist_json_name = meta_yaml["feedstock_name"]
+        logger.debug(
+            "The compute-subdir for NVIDIA URL is %s.", nvidia_redist_json_name
+        )
```

Comment:
This assumes the `extra` section may contain the needed fields. I am not sure if this is true today. I assume it is.

Reply:
The purpose of this logic is to allow controlling the redist url from inside the feedstock instead of maintaining that information here in the tick-bot implementation. Right now, none of the feedstocks will have `extra/redist-json-name` or `extra/compute-subir`; they are not expected to have them. These extra are only needed if the feedstock name is not correlated with the URL of the NVIDIA redist; e.g. when the URL of the NVIDIA redist is irregular.